src/webgpu/web_platform/reftests/canvas_colorspace.html.ts

```diff
@@ -0,0 +1,71 @@
+import { kUnitCaseParamsBuilder } from '../../../common/framework/params_builder.js';
+import { kCanvasAlphaModes, kCanvasColorSpaces } from '../../capability_info.js';
+
+import { runRefTest } from './gpu_ref_test.js';
+
+function bgra8UnormFromRgba8Unorm(rgba8Unorm: Uint8Array) {
+  const bgra8Unorm = rgba8Unorm.slice();
+  for (let i = 0; i < bgra8Unorm.length; i += 4) {
+    [bgra8Unorm[i], bgra8Unorm[i + 2]] = [bgra8Unorm[i + 2], bgra8Unorm[i]];
+  }
+  return bgra8Unorm;
+}
+
+export function runColorSpaceTest(format: GPUTextureFormat) {
+  runRefTest(async t => {
+    const device = t.device;
+
+    // prettier-ignore
+    const kRGBA8UnormData = new Uint8Array([
+      0, 255, 0, 255,
+      117, 251, 7, 255,
+      170, 35, 209, 255,
+      80, 150, 200, 255,
+    ]);
+    const kBGRA8UnormData = bgra8UnormFromRgba8Unorm(kRGBA8UnormData);
+    const width = kRGBA8UnormData.length / 4;
+
+    const testData: { [id: string]: Uint8Array } = {
+      rgba8unorm: kRGBA8UnormData,
+      bgra8unorm: kBGRA8UnormData,
+    };
+
+    function createCanvas(
+      alphaMode: GPUCanvasAlphaMode,
+      format: GPUTextureFormat,
+      colorSpace: PredefinedColorSpace
+    ) {
+      const canvas = document.createElement('canvas');
+      canvas.width = width;
+      canvas.height = 1;
+      const context = canvas.getContext('webgpu') as GPUCanvasContext;
+      context.configure({
+        device,
+        format,
+        usage: GPUTextureUsage.COPY_DST | GPUTextureUsage.RENDER_ATTACHMENT,
+        alphaMode,
+        colorSpace,
+      });
+
+      const textureData = testData[format];
+      const texture = context.getCurrentTexture();
+      device.queue.writeTexture(
+        { texture },
+        textureData.buffer,
+        {
+          bytesPerRow: textureData.byteLength,
+        },
+        { width: 4, height: 1 }
+      );
+      document.body.appendChild(canvas);
+    }
+
+    const u = kUnitCaseParamsBuilder
+      .combine('alphaMode', kCanvasAlphaModes)
+      .combine('colorSpace', kCanvasColorSpaces);
```

Comment:
nit: how about moving this out of the function and exporting it? Then it can be imported from the ref and reused directly so it's clear they need to be the same.

also nit: by convention `u` usually refers to the `unit` builder, which this is no longer.